core/src/main/java/org/elasticsearch/index/query/DisMaxQueryBuilder.java

```diff
@@ -103,6 +133,61 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     }
 
     @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        // return null if there are no queries at all
+        Collection<Query> luceneQueries = toQueries(queries, parseContext);
+        if (luceneQueries.isEmpty()) {
+            return null;
+        }
+
+        DisjunctionMaxQuery query = new DisjunctionMaxQuery(luceneQueries, tieBreaker);
+        query.setBoost(boost);
+        if (queryName != null) {
+            parseContext.addNamedQuery(queryName, query);
+        }
+        return query;
+    }
+
+    @Override
+    public DisMaxQueryBuilder readFrom(StreamInput in) throws IOException {
+        DisMaxQueryBuilder disMax = new DisMaxQueryBuilder();
+        List<QueryBuilder> queryBuilders = in.readNamedWritableList();
+        disMax.queries.addAll(queryBuilders);
+        disMax.tieBreaker = in.readFloat();
+        disMax.queryName = in.readOptionalString();
+        disMax.boost = in.readFloat();
+        return disMax;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeNamedWritableList(this.queries);
+        out.writeFloat(tieBreaker);
+        out.writeOptionalString(queryName);
+        out.writeFloat(boost);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(queries, tieBreaker, boost, queryName);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        DisMaxQueryBuilder other = (DisMaxQueryBuilder) obj;
+        return Objects.equals(queries, other.queries) &&
+               Objects.equals(tieBreaker, other.tieBreaker) &&
+               Objects.equals(boost, other.boost) &&
+               Objects.equals(queryName, other.queryName);
+    }
+
```

Comment:
yes my reasoning is that a compile error makes you think about validation rather then forgetting because there's a default empty impl that does no validation. I tend to prefer an empty validate in all queries that don't need to validate, although that's verbose. Plus that is what we do with ActionRequest as well.